src/cascadia/TerminalApp/TerminalPage.cpp

```diff
@@ -487,6 +490,85 @@ namespace winrt::TerminalApp::implementation
         }
     }
 
+    // Method Description:
+    // - This method is called when the user clicks the "export message history" button
+    //   in the query palette
+    // Arguments:
+    // - text - the text to export
+    // Return Value:
+    // - <none>
+    void TerminalPage::_OnExportChatHistoryRequested(const IInspectable& /*sender*/, const winrt::hstring& text)
+    {
+        time_t nowTime;
+        time(&nowTime);
+
+        tm nowTm;
+        localtime_s(&nowTm, &nowTime);
+
+        wchar_t buf[64];
+        wcsftime(&buf[0], ARRAYSIZE(buf), L"%F %T", &nowTm);
+
+        const auto defaultFileName = RS_(L"TerminalChatHistoryDefaultFileName") + winrt::to_hstring(buf);
+        _SaveFileHelper(text, L"", defaultFileName);
+    }
+
+    fire_and_forget TerminalPage::_SaveFileHelper(const winrt::hstring& text, const winrt::hstring& filepath, const std::wstring_view filename)
```

Comment:
Propose rename to `SaveStringToFileOrPromptUser` - this function is too complicated for what it does or what it is named.